Disable k8s components on old masters

whites11 opened the pull request:
This PR adds a new step in the Prepare phase that disables controller manager and API server in the GS masters.

whites11 committed:
Disable k8s components on old masters

whites11 committed:
Disable k8s components on old masters

whites11 committed:
Disable k8s components on old masters

whites11 committed:
Disable k8s components on old masters

whites11 committed:
Basic wiring

whites11 committed:
Basic wiring

whites11 committed:
Basic wiring

whites11 committed:
Basic wiring

whites11 committed:
Basic wiring

whites11 committed:
Basic wiring

whites11 committed:
Basic wiring

whites11 committed:
Basic wiring

calvix commented:
I don't think this will work without any PSP, as host-path volumes are not allowed by the default PSP

whites11 commented:
> I don't think this will work without any PSP, as host-path volumes are not allowed by the default PSP

I executed it on a GS cluster and it just worked. I was sure of the same thing but apparently we were wrong.

kopiczko commented on controllers/cluster_controller.go:
```diff
@@ -89,6 +97,48 @@ func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 func (r *ClusterReconciler) reconcile(ctx context.Context, cluster *capiv1alpha3.Cluster) (ctrl.Result, error) {
 	r.Log.Debugf(ctx, "calling reconcile")
+
+	migrator, err := r.MigratorFactory.NewMigrator(*cluster)
```
Feels like `NewMigrator` should take a pointer.

tuommaki replied:
+1

whites11 replied:
done

kopiczko commented on controllers/cluster_controller.go:
```diff
@@ -89,6 +97,48 @@ func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	}
+
+	if alreadyMigrated {
+		r.Log.Debugf(ctx, "cluster is already migrated")
```
This log feels pretty random when other `if`s don't emit any logs IMO.

whites11 replied:
added more logs

kopiczko commented on pkg/migration/stop_old_master.go:
```diff
@@ -0,0 +1,97 @@
+		return microerror.Mask(err)
+	}
+
+	m.logger.Debugf(ctx, "Found %d legacy nodes", len(nodeNames))
```
```suggestion
	m.logger.Debugf(ctx, "found %d legacy nodes", len(nodeNames))
```

I wouldn't be against printing names and it can help with debugging if things go south with the labels.

kopiczko commented:
PSP is there https://github.com/giantswarm/capi-migration/pull/20

But when you're testing with `make run` you use your admin credentials and I guess that can go over that.

kopiczko commented:
Re PSP: this is hanging still https://github.com/giantswarm/capi-migration/pull/22

Commit:
Update pkg/migration/stop_old_master.go

Co-authored-by: Paweł Kopiczko <pawel@giantswarm.io>

tuommaki commented on pkg/migration/azure.go:
```diff
@@ -28,10 +37,27 @@ func NewAzureMigratorFactory(cfg AzureMigrationConfig) (MigratorFactory, error)
+func (f *azureMigratorFactory) NewMigrator(cluster v1alpha3.Cluster) (Migrator, error) {
+	// Can't init the WC ctrl client here because I don't have the cluster object and so no control plane endpoint.
+
+	restConfig, err := f.config.TenantCluster.NewRestConfig(context.Background(), cluster.Name, cluster.Spec.ControlPlaneEndpoint.Host)
```
I was wondering if we need to take `cluster.Spec.ControlPlaneEndpoint.Port` into account as well or if we can just have this as-is since we are only running this towards our own clusters?

whites11 replied:
makese sense

kopiczko committed:
Extract giantswarm overlay with PSP and securityContext (#22)

Since the patch can’t be added in any other base than `default`
I decided to keep Giant Swarm specific overlay. This is PSP, namespace
and security context required to satisfy the PSP. That allows to easly
bootstrap `kubebuilder` project and add our staff on top.

Changes:

- Replace `psp` base with `giantswarm` overlay.
- Add `securityContext` patch in `giantswarm` overlay.
- Add `giantswarm` `Namespace` in `giantswarm` overlay.
- Base `dev` overlay on `giantswarm`.

whites11 committed:
Merge branch 'disable-k8s-components' of github.com:giantswarm/capi-migration into disable-k8s-components

 Conflicts:
	pkg/migration/stop_old_master.go

whites11 committed:
minor changes

tuommaki reviewed (approved):
LGTM

whites11 committed:
Make pod creation more resilient

calvix commented:
one thing that came into my mind,  this is just creating the pod but you don't know if it succeeded or if it has already been done

if there  is a step that is dependent on this  you need to somehow ensure this actually finished before running the next step

otherwise it can fuck up the workers node joining cluster

whites11 commented:
> one thing that came into my mind, this is just creating the pod but you don't know if it succeeded or if it has already been done
> 
> if there is a step that is dependent on this you need to somehow ensure this actually finished before running the next step
> 
> otherwise it can fuck up the workers node joining cluster

I actually had the code to do so, but since this pod is taking down the API server, such code was failing because it couldn't reach the API server...

calvix commented:
> I actually had the code to do so, but since this pod is taking down the API server, such code was failing because it couldn't reach the API server...

maybe it should retry until API is back?
